tests/unit/test_reddit.py

```diff
@@ -186,6 +186,26 @@ def test_submission(self):
     def test_subreddit(self):
         assert self.reddit.subreddit("redditdev").display_name == "redditdev"
 
+    @mock.patch("praw.Reddit.request", return_value=dict())
+    def test_retry(self, _):
+        self.attempts = 0
+        reddit = self.reddit
+
+        def objectify(*a, **kw):
+            if self.attempts == 0:
+                self.attempts += 1
+                raise _RetryAgain
+            else:
+                self.attempts = 0
```

Comment:
I think it would make more sense for this to just be `attempts` (not `self.attempts`), because it's a local variable for just this test. To capture the `attempts` variable inside `objectify`, add the line `nonlocal attempts` as the first line of that function's body.